e2e/nomostest/config_sync.go

```diff
@@ -618,31 +682,53 @@ func setupDelegatedControl(nt *NT) {
 		nt.T.Fatal(err)
 	}
 
-	for rsName := range nt.RootRepos {
+	// Apply all the RootSyncs
+	for id, source := range nt.SyncSources.RootSyncs() {
 		// The default RootSync is created manually, no need to set up again.
-		if rsName == configsync.RootSyncName {
+		if id.Name == configsync.RootSyncName {
 			continue
 		}
-		rs := RootSyncObjectV1Beta1FromRootRepo(nt, rsName)
+		var rs *v1beta1.RootSync
+		switch tSource := source.(type) {
+		case *syncsource.GitSyncSource:
+			rs = RootSyncObjectV1Beta1FromRootRepo(nt, id.Name)
+		case *syncsource.HelmSyncSource:
```

Comment:
Is this needed yet? It looks like you can still only specify Git `RootRepos`/`NamespaceRepos` when creating initializing NT.

Reply:
I haven't added any NT.New options for helm or oci, but they could be added later. I just put in some placeholder code for now. It also makes it reasonable to use a switch and not just an if statement, which should make for less change later.

Reply:
I feel it's a bit misleading to have this code path here even though it's currently dead/unreachable. I'd be fine with leaving this as a switch (for easy future expansion) but leave the other types (helm/oci) unimplemented

Reply:
I basically just defensively ensured all the switches check git and helm and had a default failure.

There's only 7 of these switches in the code:
- 2 in setupDelegatedControl
- 2 in setupCentralizedControl
- 2 in WatchForAllSyncs
- 1 in ValidateMultiRepoDeployments (called by WatchForAllSyncs)

I think the bottom 3 are required. But I could remove the helm case in the top 4, if you think it would avoid confusion between now and whenever we decide to add helm setup in NT.New options.

Maybe using Helm in setup doesn't makes sense. The RootSyncs all share the same RBAC, which is why most of the tests still work that switch root-sync to use Helm. But for RepoSyncs, the RBAC and Namespace are defined in the root-sync, so having the root-sync define Helm RepoSyncs should just work if we decide to add it. I'm not sure how many tests that would affect, but I didn't want to have inconsistent code that only implements the helm case for RepoSyncs...

Reply:
Looks like we have 2 tests that validate Helm RepoSyncs: TestHelmNamespaceRepo & TestHelmConfigMapNamespaceRepo. They both use central control and could be migrated to define the helm RepoSync in the NT setup. But in order for setup to sync successfully, we have to also add something like BuildAndPushHelmPackage as an NT.New option.

Reply:
I think I'm gonna hold off on that refactor till a follow up PR. It would require changing how opts.NamespaceRepos & opts.RootRepos work, to add helm and reduce them to a single map.

Reply:
Yeah agreed the bottom 3 are required but not the top 4. I'd say let's just remove the top 4 with a comment that cases must be added for helm/oci if we want to support instantiating with NT.New options.

I don't think we have enough use cases currently to warrant the added complexity, but we could always add it later.

Reply:
A compromise: I left the helm case in the but commented it out for now.